mu4/domain/importexport/internal/musicxml/importmxmlpass2.cpp

```diff
@@ -2114,6 +2117,33 @@ void MusicXMLParserPass2::measure(const QString& partId,
         measure->setBreakMultiMeasureRest(true);
     }
 
+    // measure repeat handling
+    int staffIdx = _score->staffIdx(part);
+    if (_measureRepeatsStarted) {
+        // TODO: delete anything already added to measure
```

Comment:
There are a few functions that are used for this sort of thing, but to be honest I never remember which is which.  One might be makeGap().  But what I would do is trace the code that is executed when contents of a measure are cleared for other reason, like if you select the measure and hit Delete, or if you try to enter a whole note into a measure that was already full of stuff.  Probably the latter is more relevant to your case.  Set a breakpoint on Score::setNoteRest() and single-step away!